### PR TITLE
[systemtest] Add test for switching SPS on and off

### PR DESCRIPTION
**Pull request by im-konge**
Signed-off-by: Lukas Kral <lukywill16@gmail.com>

### Type of change

- Enhancement

### Description

This PR adds new test for `StrimziPodSet`s into the `FeatureGatesIsolatedST`. It tests the ability to switching the SPS FG on and off -> components should just roll and we should be able to still communicate with brokers.

### Checklist

- [x] Write tests
- [ ] Make sure all tests pass

### CODE REVIEW

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=FeatureGatesIsolatedST

File: systemtest/src/test/java/io/strimzi/systemtest/operators/FeatureGatesIsolatedST.java [hunk 1]
```diff
@@ -242,4 +246,86 @@ public void testStrimziPodSetsFeatureGate(ExtensionContext extensionContext) {
...
+
+        LOGGER.info("Deploying CO with STS - SPS is disabled");
+
+        clusterOperator.unInstall();
+        clusterOperator = new SetupClusterOperator.SetupClusterOperatorBuilder()
+            .withExtensionContext(BeforeAllOnce.getSharedExtensionContext())
+            .withNamespace(INFRA_NAMESPACE)
+            .withWatchingNamespaces(Constants.WATCH_ALL_NAMESPACES)
+            .createInstallation()
+            .runInstallation();
```
Comment (scholzj) [inline systemtest/src/test/java/io/strimzi/systemtest/operators/FeatureGatesIsolatedST.java:275]: To make this more future proof, you should expliciely disable the FG here.

**Review by scholzj — COMMENTED** [review]
Not necessarily in this PR. But I think it would be good to have also upgrade tests which would:
* Deploy older version with SPS disabled
* Upgrade to HEAD with SPS directly enabled

**Comment by im-konge** [comment]
@scholzj I planned to have it in next PR and add it to our upgrade/downgrade tests

**Comment by scholzj** [comment]
> @scholzj I planned to have it in next PR and add it to our upgrade/downgrade tests

Sounds good.

File: systemtest/src/test/java/io/strimzi/systemtest/operators/FeatureGatesIsolatedST.java (hunk 1 above)
Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/operators/FeatureGatesIsolatedST.java:275]: Makes sense, thanks.

**Review by scholzj — APPROVED** [review]

**Pull request merged**